dGame/dBehaviors/SpawnBehavior.cpp

```diff
@@ -74,9 +76,25 @@ void SpawnBehavior::Handle(BehaviorContext* context, RakNet::BitStream* bitStrea
 		context->RegisterEndBehavior(this, branch, entity->GetObjectID());
 	}
 
-	entity->AddCallbackTimer(60, [entity] () {
-		entity->Smash();
-	});
+	// Determine if the entity spawns powerups
+	bool isPowerupSpawner = false;
```

Comment:
The logic here is special-casing powerup spawners but nothing else. That indicates to me that the "real" fix is elsewhere.

Reply:
Right, this is a bit case specific. Though it seems that the rest of the code within this function was still vital to having powerup spawners working. Without catching the edge case, the callback timer will always be added to the powerup spawner, thus destroying it after 60 seconds. 
Are there any cases where the callback is fully required? I assume probably to ensure objects are destroyed at some point, but this is a simple 60 seconds for any spawn behavior. 
The way I see it, either the handle() method needs more information to better determine how long until an object should be destroyed, or perhaps the callback should be cancelled from the powerup spawner itself. Though I don't see any way to stop a specific callback, since no handle is returned or stored, only all current callbacks on an entity can be cancelled (which is obviously problematic)

Hopefully that makes sense!

Reply:
Yeah it totally makes sense, which is why I'd want to steer this review to figuring that out. We'll probably have to wait for @Wincent01 to give you feedback here, because he and Mick built most of the scripts.